chapters/dynamic_state.md

```diff
@@ -0,0 +1,65 @@
+# Pipeline Dynamic State
+
+> [Vulkan Spec chapter](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#pipelines-dynamic-state)
+
+## Overview
+
+When creating a graphics `VkPipeline` object the logical flow for setting state is:
+
+```cpp
+// Using viewport state as an example
+VkViewport viewport = {0.0, 0.0, 32.0, 32.0, 0.0, 1.0};
+
+// Set value of state
+VkPipelineViewportStateCreateInfo viewportStateCreateInfo;
+viewportStateCreateInfo.pViewports = &viewport;
+// Create the pipeline with the state value set
+vkCreateGraphicsPipelines(viewportStateCreateInfo, &pipeline);
```

Comment:
I guess this is abbreviated pseudo-code, but even so seeing "VkPipelineViewportStateCreateInfo" as a parameter to vkCreateGraphicsPipelines looks odd. It should be "VkGraphicsPipelineCreateInfo". Maybe just add a couple more lines:
```c++
VkGraphicsPipelineCreateInfo pipelineCreateInfo;
pipelineCreateInfo.pViewportState = &viewportStateCreateInfo;
vkCreateGraphicsPipelines(pipelineCreateInfo, &pipeline);
```

Reply:
Thanks, these are all great comments and applied them all, looks much better now